tests/test-wp-title-wp-head.php

```diff
@@ -0,0 +1,241 @@
+	}
+
+	/**
+	 * Test that WP_SEO::wp_head() echoes all OG <meta> tags with expected values.
+	 *
+	 * @param  string $og_title The expected OG title content.
+	 * @param  string $og_description The expected OG description content.
+	 * @param  int    $og_image The expected OG image content.
+	 * @param  string $og_type The expected OG type content.
+	 */
+	function _assert_all_meta( $og_title, $og_description, $og_image, $og_type ) {
+		$og_img_src = wp_get_attachment_image_url( $og_image, 'og_image' );
+		$expected = <<<EOF
+<meta name='og_title' content='{$og_title}' /><!-- WP SEO -->
+<meta name='og_description' content='{$og_description}' /><!-- WP SEO -->
+<meta name='og_image' content='{$og_img_src}' /><!-- WP SEO -->
+<meta name='og_type' content='{$og_type}' /><!-- WP SEO -->
+EOF;
+		$this->assertSame( strip_ws( $expected ), strip_ws( get_echo( array( WP_SEO(), 'wp_head' ) ) ) );
+	}
+
+	/**
+	 * Wrapper for checking assert_all_meta() on OG option values.
+	 *
+	 * @param  string $key The option to test. Use a name that prefixes
+	 *     '_title', '_description', and '_keywords' in the option, like 'home'.
+	 */
+	function _assert_option_filters( $key ) {
+		$this->_assert_all_meta( $this->options[ "{$key}_og_title" ], $this->options[ "{$key}_og_description" ], $this->options[ "{$key}_og_image" ] , $this->options[ "{$key}_og_type" ] );
+	}
+
+	/**
+	 * Test default options on single page.
+	 */
+	function test_single() {
+		$this->go_to( get_permalink( $this->factory->post->create() ) );
+		$this->_assert_option_filters( 'single_post' );
+	}
+
+	/**
+	 * Test default options on custom post type.
+	 */
+	function test_singular() {
+		$this->go_to( get_permalink( $this->factory->post->create( array( 'post_type' => $this->post_type ) ) ) );
+		$this->_assert_option_filters( "single_{$this->post_type}" );
+	}
+
+	/**
+	 * Test custom meta OG on single.
+	 */
+	function test_single_custom() {
+		$post_title   = rand_str();
+		$post_excerpt = rand_str();
+		$post_id = $this->factory->post->create( array(
+			'post_title' => $post_title,
+			'post_excerpt' => $post_excerpt,
+		) );
+
+		$this->go_to( get_permalink( $post_id ) );
+		update_post_meta( $post_id, '_meta_og_title', '_custom_meta_og_title' );
+		update_post_meta( $post_id, '_meta_og_description', '_custom_meta_og_description' );
+		update_post_meta( $post_id, '_meta_og_image', $this->attachment_id );
+		update_post_meta( $post_id, '_meta_og_type', '_custom_meta_og_type' );
+		$this->_assert_all_meta( '_custom_meta_og_title', '_custom_meta_og_description', $this->attachment_id, '_custom_meta_og_type' );
+
+		update_post_meta( $post_id, '_meta_og_title', '#title#' );
+		update_post_meta( $post_id, '_meta_og_description', '#excerpt#' );
+		update_post_meta( $post_id, '_meta_og_image', $this->attachment_id );
+		update_post_meta( $post_id, '_meta_og_type', '#title#' );
+		$this->_assert_all_meta( $post_title, $post_excerpt, $this->attachment_id, $post_title );
```

Comment:
Im not sure what this line it testing.  Also, it looks like you are testing the post_title as the og_type.